core/sdk-core/src/main/java/software/amazon/awssdk/core/util/SdkUserAgent.java

```diff
@@ -40,27 +40,35 @@
     private static final String UA_BLACKLIST_REGEX = "[() ,/:;<=>?@\\[\\]{}\\\\]";
 
     /** Shared logger for any issues while loading version information. */
-    private static final Logger log = LoggerFactory.getLogger(UserAgentUtils.class);
+    private static final Logger log = LoggerFactory.getLogger(SdkUserAgent.class);
     private static final String UNKNOWN = "unknown";
+    private static volatile SdkUserAgent instance;
+
     /** User Agent info. */
-    private static volatile String userAgent;
+    private String userAgent;
+
+    private SdkUserAgent() {
+        initializeUserAgent();
+    }
+
+    public static SdkUserAgent create() {
+        if (instance == null) {
+            synchronized (SdkUserAgent.class) {
+                if (instance == null) {
+                    instance = new SdkUserAgent();
+                }
+            }
+        }
```

Comment:
Any reason to move to an instance instead of having the user just call `SdkUserAgent.getUserAgent()`?

Reply:
Yeah, I feel utility class is in general not flexible and I personally would only use it for something that is truly stateless such as calculation or string manipulation. I'd be fine if it's an internal API, and now that it's protected API, I changed it to an instance class.

Reply:
Can you give an example of something in the future that we might want to do with this that would benefit from it being an instance?

Reply:
For example, we might want to support retrieving the partial user agent string, say sdk version, via something like`sdkUserAgent.metadata().get("SdkVeresion")`. The main issue I have with this being a util class is that it holds a state and using static methods is not OOP. To me, a util class should be stateless, returning a result based on the input and nothing else.

Reply:
I usually would agree, but in this case the state that it stores is immutable, so I personally wouldn't call the class stateful. This feels closer to a class with a public static final field, except the value is loaded lazily so we have to go through a method to get to the data.

`SdkUserAgent.metadata().get("SdkVersion")` is still possible without metadata being an instance method.

The injectability could be useful, but I don't foresee us ever using it. It feels like we're just adding burden to the caller of calling create() before they call getUserAgent().

Either way, good chat! I don't think this is a blocker for me.